plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -133,25 +132,20 @@ func findNodesThatFit(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.No
 		fits := true
 		for name, predicate := range predicateFuncs {
 			fit, err := predicate(pod, node.Name, nodeNameToInfo[node.Name])
-			if err != nil {
-				switch e := err.(type) {
-				case *predicates.InsufficientResourceError:
-					if fit {
-						err := fmt.Errorf("got InsufficientResourceError: %v, but also fit='true' which is unexpected", e)
-						return api.NodeList{}, FailedPredicateMap{}, err
-					}
-				default:
-					return api.NodeList{}, FailedPredicateMap{}, err
-				}
+			if err != nil && fit {
```

Comment:
I would agree - however, this was existing code that was trying to highlight this specific case. I just tried (as best as I could) to preserve it. I would have absolutely no issues in getting rid of it. This case should never matter and the fact that the error was returned should be sufficient to ignore the other returned values.